src/schema/v2/me/conversation/conversationEventConnection.ts

```diff
@@ -0,0 +1,201 @@
+})
+
+const ConversationOrderStateChangedType = new GraphQLObjectType<
+  any,
+  ResolverContext
+>({
+  name: "ConversationOrderStateChanged",
+  interfaces: [NodeInterface],
+  description: "A conversation order state change",
+  fields: {
+    id: {
+      type: new GraphQLNonNull(GraphQLID),
+    },
+    createdAt: date,
+    state: { type: new GraphQLNonNull(GraphQLString) },
+    stateReason: { type: GraphQLString },
+  },
+})
+
+const ConversationEventType = new GraphQLUnionType({
+  name: "ConversationEvent",
+  types: [
+    MessageType,
+    ConversationOfferReceivedEventType,
+    ConversationOrderStateChangedType,
+  ],
+  resolveType: (value) => {
+    // debugger
+    const { __typename } = value
+    switch (__typename) {
+      case "CommerceOrderStateChangedEvent":
+        return "ConversationOrderStateChanged"
+      case "CommerceOfferSubmittedEvent":
+        return "ConversationOfferSubmitted"
+      default:
+        return "Message"
+    }
+  },
+})
+
+const lastMessageId = (conversation) => {
+  return get(conversation, "_embedded.last_message.id")
+}
+
+export const conversationEventConnection: GraphQLFieldConfig<
+  any,
+  ResolverContext
+> = {
+  description: "Messages and other events in a conversation.",
+  type: connectionWithCursorInfo({
+    nodeType: ConversationEventType,
+  }).connectionType,
+  args: pageable({
+    sort: {
+      type: new GraphQLEnumType({
+        name: "sortMessages",
+        values: {
+          DESC: { value: "desc" },
+          ASC: { value: "asc" },
+        },
+      }),
+    },
+  }),
+  resolve: (
+    { id, from_email, initial_message, from_name, _embedded },
+    options,
+    { conversationMessagesLoader }: { conversationMessagesLoader?: any }
+  ) => {
+    if (!conversationMessagesLoader) return null
+    const optionKeys = Object.keys(options)
+    if (optionKeys.includes("last") && !optionKeys.includes("before")) {
+      options.before = `${lastMessageId({ _embedded })}`
+    }
+    const { page, size, offset } = convertConnectionArgsToGravityArgs(options)
+    return conversationMessagesLoader({
+      page,
+      size,
+      conversation_id: id,
+      "expand[]": "deliveries",
+      sort: options.sort || "asc",
+    }).then(({ total_count, message_details }) => {
+      // Inject the convesation initiator's email into each message payload
+      // so we can tell if the user sent a particular message.
+      // Also inject the conversation id, since we need it in some message
+      // resolvers (invoices).
+      /* eslint-disable no-param-reassign */
+      const messages = message_details.map((message) => {
+        return merge(message, {
+          conversation_initial_message: initial_message,
+          conversation_from_name: from_name,
+          conversation_from_address: from_email,
+          conversation_id: id,
+        })
+      })
+      const orderEvents = orderEventStub.map((event, index) => {
+        return { ...event, id: `event-${index}` }
```

Comment:
From discussions with @lilyfromseattle 
- Would we need to filter/mutate events before returning them (example: multiple approval events) or should exchange or client do it?*
- if we add an `id` do we need to do anything else for relay cursors/pagination to work?

* Could new state changes avoid the need to filter events?